ngraph/core/reference/include/ngraph/runtime/reference/adaptive_avg_pool.hpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cfenv>
+#include <cmath>
+#include <numeric>
+#include <stdexcept>
+#include <vector>
+
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            namespace adaptive_pool
+            {
+                inline size_t window_start(size_t idx, size_t arg_shape, size_t out_shape)
+                {
+                    return floor(static_cast<double>(idx * arg_shape) / out_shape);
```

Comment:
I don't see reason cast to double and then floor from division. 
If you expect `size_t` overflow `idx` should be cast to `double`.
Calculation on `size_t` should bring the same result or I miss something?

Reply:
I should be the same, agree. The calculation in floating-point type is used to be able to use floor function to align with spec, but the result should be the same as using integer calculation.